src/main/java/net/sacredlabyrinth/phaed/simpleclans/events/CreateRankEvent.java

```diff
@@ -0,0 +1,59 @@
+package net.sacredlabyrinth.phaed.simpleclans.events;
+
+import net.sacredlabyrinth.phaed.simpleclans.Clan;
+import net.sacredlabyrinth.phaed.simpleclans.Rank;
+import org.bukkit.entity.Player;
+import org.bukkit.event.Cancellable;
+import org.bukkit.event.HandlerList;
+import org.bukkit.event.player.PlayerEvent;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * @author Minat0_
+ */
+public class CreateRankEvent extends PlayerEvent implements Cancellable {
+
+    private static final HandlerList HANDLER_LIST = new HandlerList();
+    private final String rankName;
+    private final Clan clan;
+    private boolean cancelled;
+
+    public CreateRankEvent(Player who, Clan clan, String rankName) {
+        super(who);
+        this.clan = clan;
+        this.rankName = rankName;
+    }
+
+    @NotNull
+    public String getRankName() {
+        return rankName;
+    }
+
+    public Rank getRank() {
```

Comment:
You can remove this method, as it will return null.

Reply:
Then, I can't understand the way of your minds.
CreateRankEvent, which return the String (rankName). I find it weird.
Let's imagine that CreateClanEvent will return clanName instead of `clan` object.

Reply:
Good point.
Call this event PreCreateRankEvent, as the rank doesn't exist yet.
And remove the method Rank getRank().